be/src/exec/pipeline/exchange/sink_buffer.cpp

```diff
@@ -192,7 +189,10 @@ void SinkBuffer::_process_send_window(const TUniqueId& instance_id, const int64_
     }
 }
 
-void SinkBuffer::_try_to_send_rpc(const TUniqueId& instance_id) {
+void SinkBuffer::_try_to_send_rpc(const TUniqueId& instance_id, std::function<void()> pre_works) {
```

Comment:
It's better to use lambda expression  instead of std::function

Reply:
I have tried to replace `std::function` with lambda, and I failed because there will be a problem with recursive initialization of template parameters